resources/views/components/cd/form/README.md

```diff
@@ -0,0 +1,181 @@
+# Form Components
+
+A set of form components provides support the the styled form elements in the CD Framework. 
+
+## Form
+
+The form component creates the required fieldset and legend.
+
+Use wire:submit on the form component rather than wire:click on the submit button to maintain the 
+expected behavior of submitting the form if the user hits enter in a text element.
+
+Here is an example of a simple form:
+
+```
+<x-cd.form.form legend="Simple form" wire:submit="submit">
+  <x-cd.form.text label="Name" wire:model.live="name"/>
+  <div>
+    <x-cd.form.submit-button/>
+    <x-cd.form.reset-button/>
+  </div>
+</x-cd.form.form>
+```
+
+## Form elements
+
+The `name` and `id` attributes on the form inputs default to the same value as 
+that used in the `wire:model` attribute. You may supply different values for `name` and `id  `.
+Inputs are not required by default; you can supply the `:required="true"` attribute if the input
+should be required for your form.
+
+Use the `description` attribute to provide additional instruction or formatting hints.  The description text is displayed below the input.  The component will add an `aria-describedby` attribute to the input to associate the description text with the input. 
+
+## Text
+
+The `placeholder` attribute can be used on text inputs.
+
+To adjust the width of the text input, use the `size` attribute and add the class `use-size-attr`. 
+
+Example:
+```
+    <x-cd.form.text label="Text Input" wire:model.live="name"/>
+```
+## Select
+
+The select form input requires an array of options. 
+The options and the component call are demonstrated below. The required attribute is optional.
+Making the first option disabled gives it the function of placeholder text. 
+
+```
+$this->roleoptions = [
+        [ 'value'=>'', 'option'=>'Select a Role', 'disabled'=>true],
+        [ 'value'=>'administrator', 'option'=>'Administrator', 'disabled'=>false],
+        [ 'value'=>'editor', 'option'=>'Editor', 'disabled'=>false],
+        [ 'value'=>'subscriber', 'option'=>'Subscriber', 'disabled'=>false],
+    ];
+```
+
+```
+    <x-cd.form.select :options="$roleoptions" required="1" label="Select" 
+        wire:model.live="role" />
+```
+You may add the attribute `multiple="multiple"` to create a multiselect.  Multiselect with groups are not yet supported.
+
+## Checkbox
+
+The checkbox component implements a single checkbox (see Checkboxes for multiple checkboxes).
+The checkbox component needs a value to return when the box is checked.  The checkbox-inline variant places the label next to the checkbox input rather than above it. 
+
+Note: single checkboxes cannot be required.
+
+Note: In Livewire 3, wire:model passes a boolean value to the server based on whether a checkbox is clicked or not.  If you need to pass a different (e.g., numeric) value, use wire:click instead. 
+
+```
+    <x-cd.form.checkbox label="Subscription" text="Subscribe" 
+        wire:model.live="subscribe" 
+    />
```

Comment:
Either this example should include something like a `value='1'` attribute or cd.form.checkbox should default `$value`, because this as written results in an undefined variable.